src/python/grpcio_tests/tests_aio/unit/channel_ready_test.py

```diff
@@ -0,0 +1,68 @@
+# Copyright 2020 The gRPC Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Testing the channel_ready function."""
+
+import asyncio
+import gc
+import logging
+import time
+import unittest
+
+import grpc
+from grpc.experimental import aio
+
+from tests.unit.framework.common import get_socket, test_constants
+from tests_aio.unit import _common
+from tests_aio.unit._test_base import AioTestBase
+from tests_aio.unit._test_server import start_test_server
+
+
+class TestChannelReady(AioTestBase):
+
+    async def setUp(self):
+        address, self._port, self._socket = get_socket(listen=False)
+        self._channel = aio.insecure_channel(f"{address}:{self._port}")
```

Comment:
Nit: Since https://github.com/grpc/grpc/pull/21351 has been merged, we should be able to use `LocalCredentials` now, no?

Reply:
The check for `LocalCredentials` seems not treating address `localhost` as "local". So, it will generate an `Endpoint is neither UDS or TCP loopback address.` error.

But we need `localhost` to make our unit test work for both IPv4 and IPv6.

Reply:
Hm... This seems like a bug in the `LocalCredentials` implementation. I can see the argument to be made against whitelisting `localhost` wholesale, since you can configure that to be anything in `/etc/hosts`, but I think local TCP should work with names that resolve to local addresses.

Let's gloss over this particular usage of `insecure_channel`, but we should try to address this problem in general.

@yihuazhang Thoughts?

Reply:
@lidizheng, hmm. It's weird. I am pretty sure the C-core implementation of local credentials can correctly handle the address in the format of `localhost:`. Could it be a problem of name resolver?

Reply:
@yihuazhang Good point. Since the AsyncIO stack uses its own resolver, the returned value might be unexpected to local credentials. Added a TODO issue https://github.com/grpc/grpc/issues/21940